benchmark/bench_dom_api.cpp

```diff
@@ -523,7 +523,7 @@ static void twitter_image_sizes(State& state) {
       if (not (error = tweet["entities"]["media"].get(media))) {
         for (dom::object image : media) {
           for (auto size : image["sizes"].get<dom::object>()) {
-            image_sizes.insert({ size.value["w"], size.value["h"] });
+             image_sizes.emplace(size.value["w"], size.value["h"]);
```

Comment:
@jkeiser  This gave me some headache. I do not know what ` image_sizes.insert({ size.value["w"], size.value["h"] });` does in C++. Does it create a pair of simdjson_result that then get casted to a pair of uint64_t? Anyhow, it requires that the simdjson_result value be cast to a std::pair... ?

I replaced it with something that I do understand (emplace).

Reply:
Since `image_sizes` is `vec<pair<uint64_t, uint64_t>>`, it is equivalent to `insert(std::pair<uint64_t>{ size.value["w"], size.value["h"] })`. Fully spelled out, it looks like this:

```c++
simdjson_result<dom::element> w = size.value["w"];
simdjson_result<dom::element> h = size.value["h"];
std::pair<uint64_t, uint64_t> wh{w, h};
image_sizes.insert(wh);
```

Did something break with it? I don't mind changing it at all, just want to know if the original form stopped working ... (also, there is an extra indent space in the modified line.)

I also don't understand how switching it from insert to emplace changes anything--why does emplace take two parameters and what does that mean?

Reply:
I see from the docs, it passes the varargs to the constructor. As long as the original insert form didn't break, this is totally cool. (Even if it did, this change is excellent, but we should really look into it :)

Reply:
@jkeiser The original form definitively does break, that's why it took me so long.

The orignal form requires that the simdjson_result instances be cast to std::pair instances. By making the std::pair inheritance protected, I break your code.

I did not go out into the On Demand benchmarking code because I was bored. I went there because it broke.

Reply:
@jkeiser Which is why I want this PR to be discussed with care. It *does* break your code. The fix that I provide here is *needed*. Let us be clear about that.

Reply:
To answer my own question....
```C++
image_sizes.insert({ size.value["w"], size.value["h"] });
```
in C++ means...
```
image_sizes.insert(size.value["w"]);
image_sizes.insert(size.value["h"]);
```

Reply:
I have to wonder if this actually changes the performance profile any :) Probably not much if any, honestly.